cpp/log/cert.cc

```diff
@@ -966,15 +971,20 @@ util::Status Cert::IsValidNameConstrainedIntermediateCa() const {
   // If it's not a CA cert or there is no name constraint extension then we
   // don't need to apply the rules any further
   StatusOr<bool> has_ca_constraint = HasBasicConstraintCATrue();
+  StatusOr<bool> has_name_constraints = HasExtension(NID_name_constraints);
+
   if (!has_ca_constraint.ok() || !has_ca_constraint.ValueOrDie()
-      || HasExtension(NID_name_constraints) == Cert::FALSE) {
+      || !has_name_constraints.ok() || !has_name_constraints.ValueOrDie()) {
```

Comment:
I think you're right. While it's slightly ambiguous I don't think it makes sense to return OK if the extension couldn't be looked up as it's only going to fail later when we read the data for it. That shouldn't happen anyway as it was registered at startup. We already checked the cert was loaded, which is the only other error case.